src/pages/UpSertUser/hooks/useInsertUser.ts

```diff
@@ -0,0 +1,61 @@
+import { gql } from "graphql-request";
+import { useMutation } from "react-query";
+import {
+    InsertUserMutationMutation,
+    InsertUserMutationMutationVariables,
+} from "src/generated/graphql";
+
+import useQueryClient from "src/hooks/useQueryClient";
+
+const useInsertUser = (
+    backToListPage: () => void,
+    showError: (message: string, type: "error" | "success") => void
+) => {
+    const queryClient = useQueryClient();
+    const result = useMutation<
+        InsertUserMutationMutation,
+        { response: any },
+        InsertUserMutationMutationVariables
+    >(
+        ["InsertUserMutation"],
+        async (variable) => {
+            const result = await queryClient.request<
+                InsertUserMutationMutation,
+                InsertUserMutationMutationVariables
+            >(
+                gql`
+                    mutation InsertUserMutation($email: String!, $avatar: String, $name: String!) {
+                        insert_users_one(
+                            object: { isActive: true, email: $email, avatar: $avatar, name: $name }
+                        ) {
+                            userId
+                            name
+                            email
+                        }
+                    }
+                `,
+                variable
+            );
+            return result;
+        },
+        {
+            onSuccess: () => {
+                backToListPage();
+            },
+            onError: (err) => {
+                let errorCode = err.response.errors[0].extensions.code;
+                switch (errorCode) {
+                    case "constraint-violation":
```

Comment:
please check violation on email column only

Reply:
![image](https://user-images.githubusercontent.com/58205909/131532624-518b1731-d5a6-46e1-8c09-a5bf34b545cc.png)

Reply:
This is response when insert duplicated email. I think Hasura response errors that is made by constraint of Database. But this response don't show specific column made error

Reply:
Okay